InpaintHTTP/MainMod.cs

```diff
@@ -1,111 +1,180 @@
+                    Console.WriteLine("[" + DateTime.Now.ToString("HH:mm:ss.fffff") + "] Incomming request from " + this.Request.UserHostAddress);
+                    if (this.Request.Files.Count() < 2)
                     {
-                        BitmapImg = tempBitmap;
+                        Console.WriteLine("[" + DateTime.Now.ToString("HH:mm:ss.fffff") + $"] Error, {this.Request.Files.Count()} files found");
+                        return "Err";
                     }
-                    else if (BitmapMask == null)
+
+                    Bitmap BitmapImg = null, BitmapMask = null;
+                    var donors = new List<ZsImage>();
+                    foreach (var file in this.Request.Files)
                     {
-                        BitmapMask = tempBitmap;
+                        Bitmap tempBitmap;
+                        byte[] ByteImg = new byte[file.Value.Length];
+                        file.Value.Read(ByteImg, 0, (int)file.Value.Length);
+                        using (MemoryStream ms = new MemoryStream(ByteImg))
+                            tempBitmap = new Bitmap(ms);
+
+                        if (BitmapImg == null)
+                        {
+                            BitmapImg = tempBitmap;
+                        }
+                        else if (BitmapMask == null)
+                        {
+                            BitmapMask = tempBitmap;
+                        }
+                        else
+                        {
+                            donors.Add(tempBitmap.ToArgbImage());
+                        }
                     }
-                    else {
-                        donors.Add(tempBitmap.ToArgbImage());
+
+                    var imageArgb = ConvertToArgbImage(BitmapImg);
+                    var markupArgb = ConvertToArgbImage(BitmapMask);
+
+                    var inpainter = new Inpainter();
+
+                    // NOTE: This should no longer be used since the creation of InpaintSettings comes with default variables itself?
+                    //var settings = new InpaintSettings
+                    //{
+                    //    MaxInpaintIterations = 15,
+                    //    PatchDistanceCalculator = ImagePatchDistance.Cie76,
+                    //    PatchSize = 11
+                    //};
+
+                    // Convert body request to settings
+                    InpaintSettings userSettings = new InpaintSettings();
+                    try
+                    {
+                        userSettings = JsonConvert.DeserializeObject<InpaintSettings>(Request.Body.AsString());
                     }
-                }
+                    catch { }
 
-                var imageArgb = ConvertToArgbImage(BitmapImg);
-                var markupArgb = ConvertToArgbImage(BitmapMask);
+                    // NOTE: Now merge then? giving priority to the default one?
+                    if (ApiSettings._Instance == null)
```

Comment:
This code looks a bit cryptic to a reader(e.g. "why you call a ctor without assigning an object?") It is better to implement the `Instance` as a thread safe singleton with a lazy initialization.

Reply:
This may look a bit weird indeed, instead, I will initialize ApiSettings._Instance at startup.